lib/datadog/statsd/connection.rb

```diff
@@ -12,6 +12,7 @@ def reset_telemetry
         telemetry.reset
       end
 
+      # not thread safe: callers from multiple threads must synchronize calls to this method
       def write(payload)
```

Comment:
Minor: Since we don't expect customers to use this class directly, but only through a `Sender` I suggest phrasing this from that POV, as otherwise without context it may be hard to understand what are the consequences of this important detail.

```suggestion
      # not thread safe: `Sender` instances that use this are required to properly synchronize or sequence calls to this method
      def write(payload)
```